src/AdoNetCore.AseClient/Internal/TypeMap.cs

```diff
@@ -165,16 +166,26 @@ private static int GetBinaryFormatLength(AseParameter parameter, Encoding enc, o
             }
         }
 
-        public static TdsDataType GetTdsDataType(DbType dbType, bool dbTypeIsKnown, object value, int? length, string parameterName)
+        public static AseDbType InferType(AseParameter parameter)
         {
-            // If the consumer has explicitly set a type, then rely on that.
-            if (dbTypeIsKnown && DbToTdsMap.TryGetValue(dbType, out var result))
+            if (parameter.AseDbType != AseDbType.Unsupported || parameter.Value == null || parameter.Value == DBNull.Value)
             {
-                return result(value, length ?? 0);
+                return parameter.AseDbType;
+            }
+
+            var netType = parameter.Value.GetType();
+            if (NetTypeToAseDbTypeMap.ContainsKey(netType))
+            {
+                return NetTypeToAseDbTypeMap[netType];
             }
 
-            // If that is not set, then we should try to infer the type;
-            if (NetTypeToTdsMap.TryGetValue(value.GetType(), out result))
+            throw new NotSupportedException($"Unsupported .net type {parameter.Value.GetType()} for parameter '{parameter.ParameterName}'.");
```

Comment:
The use of a NotSupportedException where previously a default value was returned looks like a breaking change. Is it fair to say that this exception will only be raised where previously the default value would have failed if sent to the server?